Recent statements all v2

amyyq2 opened the pull request:
All commits for recent statements implementation. Using to test build for UI PR

<img width="1497" alt="image" src="https://user-images.githubusercontent.com/54999459/208954968-89c30889-4631-4333-8717-dfe1af3f6879.png">

amyyq2 committed:
sql, server: implement and write to new recent statements cache

This change implements and tests a new RecentStatementsCache that is used to
store the recent statements that were executed. The cache has two new cluster
settings that tune the capacity and time that a statement lives in the cache.

This change also implements writing ActiveQueries to the cache.
The RecentStatementsCache is added to the ExecutorConfig, and the
Server writes to this cache whenever a non-internal statement finishes.
This change also removes queryMeta.phase, and replaces it with
ActiveQuery.Phase. The phases Canceled, Timed Out, Completed, and Failed are
also as possible values to ActiveQuery.Phase.

Part Of #86955
Fixes #91556

Release note: None

amyyq2 committed:
server, sql: add ListRecentStatements api and virtual table

This change implements the ListRecentStatements API that
returns all of the statements in the Recent Statements
Cache. A call to this API populates two new virtual tables:
cluster/node_recent_statements.

This PR must be merged after the recent statements cache PR:
https://github.com/cockroachdb/cockroach/pull/93270.

A future PR will implement reading from these virtual tables
in order to display recent statement info on the UI.

Part Of #86955

Release note (sql change): Added the ListRecentStatements API and
cluster/node_recent_statements tables. These contain information on
recently completed statements.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94013)
<!-- Reviewable:end -->

amyyq2 committed:
ui: add recent statements to sql activity page
This change implements reading from the recent_statements
virtual tables using sql over http. It then adds the info
from the tables to redux, and displays it on the SQL Activity
Page under a Recent Executions tab.

This PR must be merged after the recent statements cache
and virtual tables PRs: #93270 and #93927.

Part Of #86955

Release note (ui change): Adds recent statements to the
SQL Activity Page on DB Console.

amyyq2 commented:
Closing because this was used to ensure build passes: new PR is here: https://github.com/cockroachdb/cockroach/pull/94159
